### ISSUE TITLE
[Bug]: texworks issues, window management, probably related to qt.qpa.xcb or shm.

### ISSUE THREAD

**Issue by rallg** [termux/termux-packages#11992]
### Problem description

The new package `texworks` has issues, related to `qt.qpa.xcb`.
The program launches in xfce4 (Android 12, 32-bit ARM).
It works, but when a document is compiled, and the PDF window appears,
the program immediately crashes. Both the editor and PDF windows disappear.

Here is the terminal output, only the errors:
```
qt.qpa.xcb: xcb_shm_create_segment() failed for size 4414464
qt.qpa.xcb: xcb_shm_create_segment() failed for size 1582592
qt.qpa.xcb: xcb_shm_create_segment() failed for size 147264
qt.qpa.xcb: xcb_shm_create_segment() failed for size 4169968
qt.qpa.xcb: xcb_shm_create_segment() failed for size 910080
qt.qpa.xcb: xcb_shm_create_segment() failed for size 4187600
# friendly info deleted here.
Segmentation fault
```

I have not previously seen those `qt.qpa.xcp` issues using `texworks`
built without qtscript, from texworks-master code. So, after seeing
the above errors from the Termux package, I removed it, re-built the
texworks-master code as before (without qtscript), and installed it.
This also produces `qt.qpa.xcb` errors, but the program does not crash when
the PDF window appears, and the program closes without segmentation fault.

Note: With my own build, which does not use qtscript, it does not matter
whether or not the new package `qt5-qtscript` is intalled. Same massages,
same bhavior (as expected). I conclude that the problem does not involve
`qt5-qtscript` or `texworks` itslef, but rather invoplves some other qt
or xcb package.

### What steps will reproduce the bug?

Launch ltexworks, either from command line or gui.

### What is the expected behavior?

My own build, as of a few months ago, did not produce the error messages on command line, and did not have window management issues. Apparently the problem is with some support program change, rather than texworks itself.

### System information

termux-info:

```
just a moment
```

**Comment by rallg** [comment]
termux-info:

Termux Variables:
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.117
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://plug-mirror.rcac.purdue.edu/termux/termux-main stable main
# x11-repo (sources.list.d/x11.list)
deb https://plug-mirror.rcac.purdue.edu/termux/termux-x11 x11 main
Updatable packages:
All packages up to date
termux-tools version:
1.29.2
Android version:
11
Kernel build information:
Linux localhost 4.4.177-22626479 #1 SMP PREEMPT Fri Jul 22 16:53:55 +07 2022 armv8l Android
Device manufacturer:
samsung
Device model:
SM-T510

**Comment by rallg** [comment]
I have investigated further. Apparently, the problem is with `qt.qpa.xcb`.
Each time a window is opened (including dialog boxes within an application)
a message like this appears:
```
qt.qpa.xcb: xcb_shm_create_segment() failed for size <number>
```

This applies to various qt5 programs including `texworks`, `qt5ct`,
and `otter-browser`. I have no other qt5 program installed.

With `texworks`, it seems to be the case that the program attempt to re-open
(rather than refresh) the editor window, at the time the pdf window is opened. I am not sure.
Then crash with segfault, if using the Termux application.
No segfault if using my own compiled version, but the editor window does seem
to hesitate (possibly due to re-open instead of refresh).

With `otter-browser` this message appears sometime after the program is launched,
possibly at program close:
```
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1145, resource id: 4198534, major code: 40 (TranslateC>
```

So the problem has something to do wih qt, or qpa, or xcb.
I did not see this behavior until recently (last few months).

**Comment by rallg** [comment]
More: Just today, there was an update to package `poppler-qt`. That seems to be the kind of program that might be involved in the crash and segfault. So after installing the update, I tried `texworks` again, and now there is no crash or segfault. Whether `poppler-qt` fixed it, or just lucky, I do not know.

Also: the `tq.qpa.xcb` messages seem to be very common, according to various online message boards concerning qt applications. Apparently they are a harmless nuisance.

In summary, the issue is possibly fixed. Closing it.
